ApollosProject/apollos-prototype
# Fixes current campus not showing "distance to location"

## DESCRIPTION

### What does this PR do, or why is it needed?

There were two issues with the existing code that this PR fixes. 

- The `sortedCampuses` logic (that drives the list of campuses in teh screen) was pulling a campus object that didn't have the `distance` attached (because it came from the currentUser, instead of `campusesQuery`. A use of `find` fixes that. 
- Querying the `distanceToCampus` field not under the context of the `campuses` field results in a `null` value. That field having different values in different contexts confuses apollos cache resolution, and so the wrong value is cached. Thus, we need to only query for `distanceToLocation` when that field will contain the correct value.

### What design trade-offs/decisions were made?
n/a


### How do I test this PR?
1. Visit the location finder screen.
2. Confirm that your current location has a distance on the tile
3. Change your location, and revisit, confirming that your current location updated correctly.

### What automated tests did you write?


## TODO:


## REVIEW:

**Manual QA**


**Code Review: Questions to consider**


> The purpose of PR Review is to _improve the quality of the software._
